101-custom-rp-with-function/README.md

```diff
@@ -7,89 +7,67 @@
+This sample deployment creates the following apis on the resource.
 
-1) An ARM extended resource called "users"
-2) An API called "ping"
+1) An Azure Resource Manager extended API called "customResources"
 
-### Users
+### CustomResources
 
-The users resource is defined in the following part of the ARM template : 
+"customResources" is an API extension contained within the custom resource provider. This extension is defined in the following part of the ARM template under the "resourceProviders" resource:
 
 ```json
 "resourceTypes": [
-                                {
-                                    "name":"users",
-                                    "routingType":"Proxy,Cache",
-                                    "endpoint": "[concat('https://', parameters('funcname'), '.azurewebsites.net/api/{requestPath}')]"
-                                }
-                            ]
+                    {
+                        "name": "customResources",
+                        "routingType": "Proxy",
+                        "endpoint": "[listSecrets(resourceId('Microsoft.Web/sites/functions', parameters('funcName'), 'HttpTrigger1'), '2018-02-01').trigger_url]"
+                    }
+                 ]
 ```
 
-In the above template we can see that the resource "users" has been defined as a Proxy resource which means that it is backed by a REST api endpoint. The details of the endpoint come next. The endpoint uri in the section refers to the endpoint that implements this resource. When the resource is defined as above , the resource will support all CRUD calls like "GET", "PUT", "DELETE" etc and it is expected that the endpoint has implemented them. In the above case this means that we will be able to make the following calls on ARM:
+In the above template we can see that the "customResources" API has been defined as a Proxy resource which means that it is backed by a RESTful API endpoint. The details of the endpoint come next. The endpoint uri in the section refers to the endpoint that implements this resource. When the resource is defined as above , the resource will support all CRUD calls like "GET", "PUT", "DELETE" etc and it is expected that the endpoint has implemented them. In the above case, this means that we will be able to make the following calls on Azure Resource Manager:
 
 ```
-PUT/GET/DELETE /subscriptions/{subscriptionid}/resourceGroups/{resourcegroup}/providers/Microsoft.CustomProviders/resourceProviders/{customrpname}/users/name?<api-version>
+PUT/GET/DELETE /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomProviders/resourceProviders/{resourceProviderName}/customResources/{customResourceName}?api-version=2018-09-01-preview
 ```
 
-You can see this in action in the last part of the template where we create this new user type : 
+The last defined resource in the Azure Resource Manager template utilizes the new Azure extension to create an instance of the "customResources": 
```

Comment:
Can we reword this - if we change the template so this is not the last resource...